skargo/src/manifest.sk

```diff
@@ -1,43 +1,28 @@
 module Skargo;
 
-class Version(major: Int, minor: Int, patch: Int) uses Orderable {
-  static fun fromString(str: String): Version {
-    components = str.split(".");
-    if (components.size() != 3) {
-      invariant_violation("Invalid version string " + str)
-    };
-    Version(components[0].toInt(), components[1].toInt(), components[2].toInt())
-  }
-
-  fun toString(): String {
-    `${this.major}.${this.minor}.${this.patch}`
-  }
-}
-
-base class VersionConstraint uses Equality {
-  children =
-  | RangeConstraint(low: Version, high: ?Version)
-  | PathConstraint(path: String, version: ?Version)
-
-  static fun fromTOML(basePath: String, toml: TOML.Value): VersionConstraint {
+// TODO: Add support for `git` repositories as a package location.
+class ManifestDependency{
+  version: ?Semver.VersionReq = None(),
+  path: ?String = None(),
+} {
+  static fun fromTOML(basePath: String, toml: TOML.Value): ManifestDependency {
     toml match {
-    | TOML.String(v) -> static::fromString(v)
+    | TOML.String(v) ->
+      ManifestDependency{version => Some(Semver.VersionReq::fromString(v))}
     | t @ TOML.Table _ ->
       t.maybeGetString("path") match {
       | Some(path) ->
-        PathConstraint(
-          Path.join(basePath, path),
-          t.maybeGetString("version").map(Version::fromString),
-        )
+        ManifestDependency{
+          // FIXME: This breaks on absolute paths.
+          path => Some(Path.join(basePath, path)),
+          version => t
+            .maybeGetString("version")
+            .map(Semver.VersionReq::fromString),
+        }
```

Comment:
Fail if `t` has non-`path`, non-`version` fields?

Reply:
Yes we should somehow validate the schema of the whole file.